Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.mm

```diff
@@ -0,0 +1,650 @@
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if !__has_feature(objc_arc)
+#error This file requires ARC. Add the "-fobjc-arc" compiler flag for this file.
+#endif
+
+#import "config.h"
+#import "_WKWebExtensionContextInternal.h"
+
+#import "CocoaHelpers.h"
+#import "WebExtension.h"
+#import "WebExtensionContext.h"
+#import "_WKWebExtensionControllerInternal.h"
+#import "_WKWebExtensionInternal.h"
+#import "_WKWebExtensionMatchPatternInternal.h"
+#import "_WKWebExtensionTab.h"
+#import <WebCore/WebCoreObjCExtras.h>
+
+NSNotificationName const _WKWebExtensionContextPermissionsWereGrantedNotification = @"_WKWebExtensionContextPermissionsWereGranted";
+NSNotificationName const _WKWebExtensionContextPermissionsWereDeniedNotification = @"_WKWebExtensionContextPermissionsWereDenied";
+NSNotificationName const _WKWebExtensionContextGrantedPermissionsWereRemovedNotification = @"_WKWebExtensionContextGrantedPermissionsWereRemoved";
+NSNotificationName const _WKWebExtensionContextDeniedPermissionsWereRemovedNotification = @"_WKWebExtensionContextDeniedPermissionsWereRemoved";
+
+NSNotificationName const _WKWebExtensionContextPermissionMatchPatternsWereGrantedNotification = @"_WKWebExtensionContextPermissionMatchPatternsWereGranted";
+NSNotificationName const _WKWebExtensionContextPermissionMatchPatternsWereDeniedNotification = @"_WKWebExtensionContextPermissionMatchPatternsWereDenied";
+NSNotificationName const _WKWebExtensionContextGrantedPermissionMatchPatternsWereRemovedNotification = @"_WKWebExtensionContextGrantedPermissionMatchPatternsWereRemoved";
+NSNotificationName const _WKWebExtensionContextDeniedPermissionMatchPatternsWereRemovedNotification = @"_WKWebExtensionContextDeniedPermissionMatchPatternsWereRemoved";
+
+_WKWebExtensionContextNotificationUserInfoKey const _WKWebExtensionContextNotificationUserInfoKeyPermissions = @"permissions";
+_WKWebExtensionContextNotificationUserInfoKey const _WKWebExtensionContextNotificationUserInfoKeyMatchPatterns = @"matchPatterns";
+
+@implementation _WKWebExtensionContext
+
+#if ENABLE(WK_WEB_EXTENSIONS)
```

Comment:
Nit: Should all of the constants be in the `ENABLE` block? Or do they not need to be?

Reply:
They need to be exported symbols, since the header is not behind a guard since guards can't be in the public headers.